hikvision_tftpd.py

```diff
@@ -34,20 +35,20 @@ class Server(object):
     _TFTP_OPCODE_ACK = 4
     _TFTP_OPCODE_OACK = 6
     _TFTP_ACK_PREFIX = struct.pack('>h', _TFTP_OPCODE_ACK)
-    BLOCK_SIZE = 512
 
     def __init__(self, handshake_addr, tftp_addr, filename, file_contents):
         self._file_contents = file_contents
         self._filename = filename
-        self._total_blocks = ((len(file_contents) + self.BLOCK_SIZE)
-                              // self.BLOCK_SIZE)
+        self._block_size = _DEFAULT_BLOCK_SIZE
```

Comment:
could lines 42, 43, and 51 be just `self._set_block_size(_DEFAULT_BLOCK_SIZE)` to reduce code duplication?

Reply:
Same issue with the max blocks error check. This could be self._set_default_block_size() though...